Try to unify how documentation is shown in all platforms (#122)
undefinedlabs/scope-docs: docs/dotnet-configuration.md

sidebar_label: Configuration
---

> The behaviour of the Scope Agent can be modified by using  environment variables or by using a configuration file `Scope.yml`in the source root of the project.

Reviewer: ```suggestion
The behaviour of the Scope Agent can be modified by using environment variables or a configuration file `Scope.yml` in the source root of the project.
```

> These configuration settings can also be provided via CLI flags. Run `scope-run --help` for more information.

## Disabling Scope instrumentation

You can disable Scope auto instrumentation. Scope auto instruments your code and tests by default, you can disable this behaviour.

### Using Environment Variables

```sh

> SCOPE_AUTO_INSTRUMENT=False

Reviewer: don't you need to export? Or inline this with your command. Otherwise this does nothing, right?

Author: yes, but every platform (windows, Mac, linux,...) has a different way to set Env variables, so I thought that showing g export explicitly was not clear, also GitHub Action has a different way to set them

Reviewer 2: I'd show only `SCOPE_AUTO_INSTRUMENT=false`. Not always needed to use commands, e.g. GitHub Actions.

Reviewer 3: That variable doesn't exist in the dotnet agent anymore, the name is `SCOPE_INSTRUMENTATION_ENABLED` to follow the same pattern as the `scope.yml`:
```yaml
scope:
    instrumentation:
        enabled: true
```

Reviewer 3: maybe we need to review all env vars, but should match the `scope.yml` file

Author: Changed for .net

Reviewer 3: Env vars should be standard for all agents isn't?

```

### Using YAML Configuration file

```yaml
scope:

>   auto_instrument: false

Reviewer: wasn't this 
```yaml
scope:
  instrumentation:
    enabled: true
```
or is this different?

Author: I will check, I copied it from some other platform

Reviewer 2: In Java is:
```
scope:
  instrumentation:
    enabled: true
```

Author: Changed to:
 scope:
  instrumentation:
    enabled: false

  
```

## Changing service name

> You can change the name of the service when sending data to Scope. Scope initially uses `default` for the service name.

Reviewer: ```suggestion
You can specify the name of the service when sending data to Scope. Scope uses `default` as fallback.
```
though a bit redundant, you could also say `default` as default

### Using Environment Variables

```sh

> SCOPE_SERVICE="service name"

Reviewer: same feedback as before and for all the rest

Author: Changed to "service-name"

```

### Using YAML Configuration file

```yaml
scope:
  service: 'service name'
```

## Setting Scope as Global Tracer

> You can set Scope as the Global Tracer of Open Tracing. Scope uses Open Tracing internally to handle Span and traces, using its own tracer, You can set it as the global tracer so it will capture all the spans generated by any Open Tracing library in relation to your tests. 

Reviewer: why is `Global Tracer` capitalised? 
Also, the phrase is a bit unclear imo.
How about something like
```suggestion
Scope handles Open Tracing span and traces using its own tracer. You can set Scope as the global tracer of Open Tracing so it will capture all the spans generated by any Open Tracing library in relation to your tests. 
```

### Using Environment Variables

```sh
SCOPE_SET_GLOBAL_TRACER=True
```

### Using YAML Configuration file

```yaml
scope:
  set_global_tracer: true
  
```

## Change testing mode

> These variable indicates to the agent if it is running tests (`true`) or it is being used for runtime instrumentation (`false`).

Reviewer: ```suggestion
This variable indicates whether the agent is running tests (`true`) or being used for runtime instrumentation (`false`).
```

### Using Environment Variables

```sh
SCOPE_TESTING_MODE=False
```

### Using YAML Configuration file

```yaml
scope:
  testing_mode: false
  
```

>  By default `SCOPE_TESTING_MODE` property depends on whether the build has been triggered by a CI server (`true`), or not (`false`).

Reviewer: `SCOPE_TESTING_MODE` property? Isn't that a env variable? 

What we want to say here is probably something along these lines: 
> If `SCOPE_TESTING_MODE` is not specified, the agent sets it to true when running in CI and false otherwise.

Reviewer 3: The name changed to: `SCOPE_MODE_TESTING` to follow the `scope.yml`:
```yaml
scope:
    mode:
        testing: true
```

Author: changed

sidebar_label: Configuration
---

The behaviour of the Scope Agent can be modified using environment variables or a configuration file `scope.yml` in the source root of the project.

> These configuration settings can also be provided via CLI flags. Run `scope-run --help` for more information.

## Disabling Scope instrumentation

Scope auto instruments your code and tests by default. You can disable this behaviour.

### Using Environment Variables

```sh
SCOPE_INSTRUMENTATION_ENABLED=false
```

### Using YAML Configuration file

```yaml
scope:
  instrumentation:
    enabled: false
```

## Changing service name

You can specify the name of the service when sending data to Scope. Scope uses `default` as fallback.

### Using Environment Variables

```sh
SCOPE_SERVICE="service-name"
```

### Using YAML Configuration file

```yaml
scope:

>   service: 'service-name'

Reviewer 4: ```suggestion
  service: service-name
```

```

## Setting Scope as Global Tracer

Scope handles Open Tracing spans and traces using its own tracer. You can set Scope as the global tracer so that it captures all the spans generated by any Open Tracing library. 

### Using Environment Variables

```sh
SCOPE_TRACER_GLOBAL=True
```

### Using YAML Configuration file

```yaml
scope:
  tracer:
      global: true

>   

Reviewer 3: remove this empty line

Author: removed

```

## Change testing mode

This variable indicates whether the agent is running tests (`true`) or being used for runtime instrumentation (`false`).

### Using Environment Variables

```sh
SCOPE_TESTING_MODE=false
```

### Using YAML Configuration file

```yaml
scope:
  mode:

>     testing: false

Author: added to the points for next agent meeting

>   

Reviewer 3: same here

Author: removed

### Using Environment Variables

```sh

> SCOPE_METADATA="sample.key1=$SAMPLE_VAR1,sample.key2=$SAMPLE_VAR2,sample.key3=sampleValue3"

Reviewer 2: Please check in your agents if the `"` character is trimmed. In the Java agent, I had to fix this issue.

### Using Environment Variables

```sh

> SCOPE_CONFIGURATION="sample.key1,sample.key2"

Reviewer 2: Please check in your agents if the " character is trimmed. In the Java agent, I had to fix this issue.
